### PR TITLE
[Feature]added jwt authentication

### PR DESCRIPTION
**Pull request by arj9912**

### CODE REVIEW

File: autho/urls.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 from django.urls import path
 from . import views
+from rest_framework_simplejwt.views import TokenVerifyView, TokenObtainPairView, TokenRefreshView
```
Comment (RahulRimal) [inline autho/urls.py:3]: import from autho's view and not from library's view

File: autho/urls.py [hunk 2]
```diff
@@ -8,6 +9,13 @@
 
 router.register('users', views.UserViewSet, basename='users')
 
-urlpatterns = []
+
+urlpatterns = [
+    path("create-token/", TokenObtainPairView.as_view(), name="token_obtain_pair"),
+    path("refresh-token/",TokenRefreshView.as_view(), name="token_refresh"),
+    path("verify-token/", TokenVerifyView.as_view(), name="token_verify"),
+    # path('register/', views.register_user, name='register'),
```
Comment (RahulRimal) [inline autho/urls.py:17]: remove this

File: autho/views.py [hunk 3]
```diff
@@ -1,5 +1,12 @@
+from django.http import HttpRequest
```
Comment (RahulRimal) [inline autho/views.py:1]: fix the import order
python libraries at top
then django libraries
then rest_framework libraries
then third_party libraries
finally own custom apps imports

File: autho/views.py [hunk 4]
```diff
@@ -9,4 +16,42 @@
...
+    serializer_class = UserSerializer
+
+
+    @action(detail=False, methods = ['GET', 'PATCH'])
+    def me(self, request):
+        user = request.user
+        serializer = self.get_serializer(user)
+        return Response(serializer.data)
+
+    # def me(self, request):
```
Comment (RahulRimal) [inline autho/views.py:28]: remove commented code

**Review by RahulRimal — CHANGES_REQUESTED** [review]

File: autho/urls.py (hunk 1 above)
Comment (arj9912) [reply autho/urls.py:3]: done

File: autho/urls.py (hunk 2 above)
Comment (arj9912) [reply autho/urls.py]: done

File: autho/views.py (hunk 4 above)
Comment (arj9912) [reply autho/views.py]: done

File: autho/urls.py (hunk 1 above)
Comment (RahulRimal) [reply autho/urls.py:3]: make line 2 from . import views
and delete line 3

File: autho/urls.py [hunk 5]
```diff
@@ -1,20 +1,19 @@
...
+from . import views as autho_views
 from rest_framework_simplejwt.views import TokenVerifyView, TokenObtainPairView, TokenRefreshView
 
 # from rest_framework.routers import DefaultRouter
 from rest_framework_nested.routers import DefaultRouter, NestedDefaultRouter
 
 router = DefaultRouter()
 
-router.register('users', views.UserViewSet, basename='users')
+router.register('users', autho_views.UserViewSet, basename='users')
```
Comment (RahulRimal) [inline autho/urls.py:10]: make autho_views.UserViewSet to views.UserViewSet

File: autho/urls.py [hunk 6]
```diff
@@ -1,20 +1,19 @@
...
 
 
 urlpatterns = [
     path("create-token/", TokenObtainPairView.as_view(), name="token_obtain_pair"),
```
Comment (RahulRimal) [inline autho/urls.py:14]: make TokenObtainPairView.as_view() to views.TokenObtainPairView.as_view()

File: autho/urls.py [hunk 7]
```diff
@@ -1,20 +1,19 @@
...
     path("refresh-token/",TokenRefreshView.as_view(), name="token_refresh"),
```
Comment (RahulRimal) [inline autho/urls.py:15]: make TokenRefreshView.as_view() to views.TokenRefreshView.as_view()

File: autho/urls.py [hunk 8]
```diff
@@ -1,20 +1,19 @@
...
     path("verify-token/", TokenVerifyView.as_view(), name="token_verify"),
```
Comment (RahulRimal) [inline autho/urls.py:16]: make TokenVerifyView.as_view() to views.TokenVerifyView.as_view()

**Review by RahulRimal — CHANGES_REQUESTED** [review]

File: autho/urls.py (hunk 5 above)
Comment (arj9912) [reply autho/urls.py]: ok done

File: autho/urls.py (hunk 6 above)
Comment (arj9912) [reply autho/urls.py]: ok done

File: autho/urls.py (hunk 7 above)
Comment (arj9912) [reply autho/urls.py]: ok done

File: autho/urls.py (hunk 8 above)
Comment (arj9912) [reply autho/urls.py]: ok done

File: autho/urls.py (hunk 1 above)
Comment (arj9912) [reply autho/urls.py]: ok done

File: autho/views.py [hunk 9]
```diff
@@ -1,12 +1,69 @@
...
+    def me(self, request):
+        user = request.user
+        serializer = self.get_serializer(user)
+        return Response(serializer.data)
+    
+    
+    @action(detail=False, methods=["POST"])
+    def change_password(self, request: HttpRequest) -> Response:
+        user: User = request.user
+        old_password = request.data.get("old_password")
```
Comment (RahulRimal) [inline autho/views.py:33]: here check if request.user.is_authenticated
if not authenticated then show error saying must be logged in to change password

**Review by RahulRimal — CHANGES_REQUESTED** [review]

**Review by RahulRimal — APPROVED** [review]

**Pull request merged**